hare/committracker.go

```diff
@@ -59,35 +59,45 @@ func (ct *commitTracker) OnCommit(ctx context.Context, msg *Msg) {
 		return
 	}
 
+	msgHash := types.BytesToHash(msg.Message.HashBytes())
+	metadata := types.HareMetadata{
+		Layer:   msg.Layer,
+		Round:   msg.Round,
+		MsgHash: msgHash,
+	}
 	if prev, ok := ct.seenSenders[msg.NodeID]; ok {
-		if prev.InnerMsg.Layer == msg.Layer &&
-			prev.InnerMsg.Round == msg.Round &&
-			prev.InnerMsg.MsgHash != msg.MsgHash {
-			ct.logger.WithContext(ctx).With().Warning("equivocation detected in commit round",
```

Comment:
i honestly feel that this condition capture the essence of malfeasance behavior in a compact form:
the same signer signs two messages for the same layer/round

Reply:
I don't like the verbosity of this check, it is also used in multiple places in the same way. I created a method on `HareMetadata` `Equivocation` that contains this check so instead of:

```golang
if prev.InnerMsg.Layer == msg.Layer &&
	prev.InnerMsg.Round == msg.Round &&
	prev.InnerMsg.MsgHash != msg.MsgHash {
```

one can write

```golang
if prev.InnerMsg.Equivocation(&metadata) {
```

instead. This is also allows for different equivocation checks (if we change them at some point) to be neatly contained within this method.